ssl/ssl_transfer_asn1.cc

```diff
@@ -1066,16 +1006,12 @@ static int SSL_parse(SSL *ssl, CBS *cbs, SSL_CTX *ctx) {
   int quiet_shutdown;
   int ssl_config_present = 0;
 
-  if (!CBS_get_asn1(cbs, &ssl_cbs, CBS_ASN1_SEQUENCE) ||
-      CBS_len(cbs) != 0 ||
-      !CBS_get_asn1_uint64(&ssl_cbs, &ssl_serial_ver)) {
+  if (!CBS_get_asn1(cbs, &ssl_cbs, CBS_ASN1_SEQUENCE) || CBS_len(cbs) != 0 ||
+      !CBS_get_asn1_uint64(&ssl_cbs, &ssl_serial_ver)
+      || ssl_serial_ver > SSL_SERIAL_VERSION) {
```

Comment:
Won't this fail if you attempt to parse a v2 formatted object with a v1 parser?

Reply:
This is just matching with what we had in mainline by applying the commits. I included you because your change was previously reviewed and included here, and that change only made the SSLConfig object itself more nimble for parsing, and not this top-level version attribute. It needs to be fixed in main first if that's what we want to do, and not here.